accepted/2021/package-validation/apicompat.md

```diff
@@ -0,0 +1,143 @@
+
+### 3. Cannot make member non virtual
+This rule checks that a member should be virtual on both the implementation and in the contract when the member is overridable:
+1. Is virtual.
+2. Is not sealed.
+3. Type is not sealed.
+
+### 4. Cannot remove base type or interface
+This rule should check that the base types and interfaces match on both the implementation and in the contract.
+
+### 5. Cannot seal type
+This rule checks if the type is non inheritable on both the implementation and contract:
+1. Has the sealed modifier
+2. Has no non static visible constructors outside the assembly.
+
+### 6. Enum type must match
+Enum underlying type must match on both the implementation and contract.
+
+### 7. Enum values must match
+Enum values most match on both the implementation and contract.
+
+### 8. Interface should have same members
+Interface members must match on both the implementation and contract.
+
+We should have a rule setting that users can set to support allowing adding a new member to the interface if the member includes a default implementation. Adding a member to an interface on new releases is a breaking change unless the member has a default implementation, this requires C# 8.0.
+
+### 9. Members must exist
+This rules checks that all the public members should exist on both the implementation and contract. This applies to any member that can be accessible within outside of the assembly, excluding interface members as those are handled in a separate rule.
+
+1. This rules makes sure that the member exists on the type or any of it's base types.
+2. Parameter types are equal.
+3. Return types match.
+
+### 10. Modifiers cannot change
+This rule checks for differences in marshaling attributes like `in`, `out`, `ref`, `readonly` and custom modifiers like `const` and `volatile` in parameters and member return's type.
+
+1. Return's type modifiers can be ref, readonly or none. 
+2. Parameter modifiers can be in, out, ref and custom modifiers like const and volatile can exist.
+
+### 11. Parameter names cannot change
+This rule makes sure that constructor, property or method parameter names match.
+
+### 12. Type cannot change classification
+This rule makes sure that type's classification doesn't change. Type classification refers to, `class`, `struct`, `ref struct`, `readonly`, `interface` or `delegate`.
+
+### 13. Nullable annotations should match
+This rule should make sure that nullable annotations match in between the implementation and the contract.
+
+Also, this rule should allow differences in nullable annotations that are not breaking, for example, relaxing a nullable reference type annotation on an in parameter (i.e `public string Foo(string a) -> public string Foo(string? a)`). However, this should be configurable via rule settings to disable this as in some cases where you are comparing a reference assembly vs it's implementation you want them to match in all cases.
```

Comment:
Does the above list cover everything relevant called out in the breaking change guidance we have in dotnet/runtime?

Reply:
Also, are there any differences between this and what apicompat protects in dotnet/runtime?

Reply:
Yes, APICompat doesn't yet support nullable.  I think we should be calling out all the rules which APICompat has today as well as those which we deem are needed.

Reply:
I doubled checked and this list covers everything we have in dotnet/runtime. It is a super set as @ericstj mentioned we don't yet support nullable in APICompat.